src/accessors-geog.cpp

```diff
@@ -0,0 +1,29 @@
+#include <s2geography.h>
+
+#include "geography.hpp"
+#include "pybind11.hpp"
+
+namespace py = pybind11;
+namespace s2geog = s2geography;
+using namespace spherely;
+
+PyObjectGeography convex_hull(PyObjectGeography a) {
+    const auto& a_ptr = a.as_geog_ptr()->geog();
+    auto s2_obj = s2geog::s2_convex_hull(a_ptr);
+    auto geog_ptr = std::make_unique<Geography>(std::move(s2_obj));
```

Comment:
So currently, this function returns instances of the base class `Geography`. If I change this to `std::make_unique<Polygon>`, then it actually returns a `Polygon` python subclass. 

For the convex_hull case, I think I can be _sure_ that this will always return a Polygon. But in general, do we need a method on `s2geography::Geography` to know which subtype it actually is (a PolygonGeography) in this case), so we can assert / known to which subclass to downcast on our side?

Reply:
I don't know if this is the same for s2, but with GEOS there are certainly geometry-returning functions that can return varying geometry types depending on the input (eg an intersection can basically return any geometry type)

Reply:
> I don't know if this is the same for s2, but with GEOS there are certainly geometry-returning functions that can return varying geometry types depending on the input (eg an intersection can basically return any geometry type)

If you know the type of the input(s) can you always determine the type of the output(s), or is there some geometry-returning functions where the output geometry type depends on some internal processing logic?

In the 1st case, we could use `spherely::Geography::geog_type()` returned from the inputs. Although I agree this might be worth of having such method available in s2geography, which would help getting rid of `spherely::Geography` wrapper classes that makes things quite confused.

Reply:
> If you know the type of the input(s) can you always determine the type of the output(s), or is there some geometry-returning functions where the output geometry type depends on some internal processing logic?

Yes, AFAIK that is the case. For example consider intersection of two polygons: if they overlap, you can get a Polygon or MultiPolygon, if they touch along a segment you get a LineString (or MultiLineString), and if they touch you get a Point (or MultiPoint), and if they both overlap in some area and touch in another area, you get a GeometryCollection.